src/Host/Client/Impl/Host/RemoteBrokerClient.cs

```diff
@@ -2,55 +2,76 @@
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
 using System;
+using System.Diagnostics;
+using System.Net;
 using System.Net.Http;
-using System.Net.NetworkInformation;
-using System.Net.Sockets;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Common.Core;
 using Microsoft.Common.Core.Logging;
+using Microsoft.Common.Core.Net;
 using Microsoft.R.Host.Client.BrokerServices;
 
 namespace Microsoft.R.Host.Client.Host {
     internal sealed class RemoteBrokerClient : BrokerClient {
-        public RemoteBrokerClient(string name, Uri brokerUri, IntPtr applicationWindowHandle, IActionLog log)
-            : base(name, brokerUri, brokerUri.Fragment, new RemoteCredentialsDecorator(applicationWindowHandle, brokerUri), log) {
+        private readonly IntPtr _applicationWindowHandle;
+        private readonly string _authority;
+        private readonly ICredentialsDecorator _credentials;
+        private int _certificateUIActive;
 
+        static RemoteBrokerClient() {
+            ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+        }
+
+        public RemoteBrokerClient(string name, Uri brokerUri, IActionLog log, IntPtr applicationWindowHandle)
+            : base(name, brokerUri, brokerUri.Fragment, log, applicationWindowHandle) {
+            _applicationWindowHandle = applicationWindowHandle;
+
+            _credentials = new RemoteCredentialsDecorator(applicationWindowHandle, brokerUri);
+            _authority = new UriBuilder { Scheme = brokerUri.Scheme, Host = brokerUri.Host, Port = brokerUri.Port }.ToString();
             CreateHttpClient(brokerUri);
         }
 
         public override string HandleUrl(string url, CancellationToken ct) {
             return WebServer.CreateWebServer(url, HttpClient.BaseAddress.ToString(), ct);
         }
 
+        protected override ICredentialsDecorator Credentials => _credentials;
```

Comment:
Sure. It was needed since it used to be NetworkCredential but since you added the decorator, it is no longer needed.

---
In reply to: [82300188](https://github.com/Microsoft/RTVS/pull/2447#discussion_r82300188) [](ancestors = 82300188)